client/my-sites/plans-v2/plans-filter-bar-i5/index.tsx

```diff
@@ -0,0 +1,139 @@
+/**
+ * External dependencies
+ */
+import React, { useMemo, useRef, useState, useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import { useTranslate } from 'i18n-calypso';
+import classNames from 'classnames';
+import { ToggleControl } from '@wordpress/components';
+import { useMobileBreakpoint } from '@automattic/viewport-react';
+
+/**
+ * Internal dependencies
+ */
+import {
+	JETPACK_PRODUCTS_BY_TERM,
+	PRODUCT_JETPACK_BACKUP_REALTIME,
+} from 'calypso/lib/products-values/constants';
+import {
+	JETPACK_RESET_PLANS_BY_TERM,
+	PLAN_JETPACK_SECURITY_REALTIME,
+} from 'calypso/lib/plans/constants';
+import isJetpackCloud from 'calypso/lib/jetpack/is-jetpack-cloud';
+import { TERM_MONTHLY, TERM_ANNUALLY } from 'calypso/lib/plans/constants';
+import { masterbarIsVisible } from 'calypso/state/ui/selectors';
+import { getJetpackCROActiveVersion } from '../abtest';
+import useDetectWindowBoundary from '../use-detect-window-boundary';
+import { getHighestAnnualDiscount } from '../utils';
+
+/**
+ * Type dependencies
+ */
+import type { FilterBarProps, DiscountMessageProps } from '../types';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+const CALYPSO_MASTERBAR_HEIGHT = 47;
+const CLOUD_MASTERBAR_HEIGHT = 94;
+
+const REALTIME_PRODUCTS = [ PLAN_JETPACK_SECURITY_REALTIME, PRODUCT_JETPACK_BACKUP_REALTIME ];
+
+const DiscountMessage: React.FC< DiscountMessageProps > = ( { primary } ) => {
+	const translate = useTranslate();
+	const isMobile: boolean = useMobileBreakpoint();
+
+	const croVersion = getJetpackCROActiveVersion();
+	const slugsToCheck = useMemo(
+		() =>
+			[
+				...JETPACK_PRODUCTS_BY_TERM.map( ( s ) => s.yearly ),
+				...JETPACK_RESET_PLANS_BY_TERM.map( ( s ) => s.yearly ),
+			].filter( ( slug ) => {
+				// Don't factor in real-time products for CRO v1 --
+				// they're not visible in the product grid
+				if ( croVersion === 'v1' && REALTIME_PRODUCTS.includes( slug ) ) {
+					return false;
+				}
+
+				return true;
+			} ),
+		[ croVersion ]
+	);
+	const highestAnnualDiscount = useSelector( ( state ) =>
+		getHighestAnnualDiscount( state, slugsToCheck )
+	);
+
+	if ( ! highestAnnualDiscount ) {
+		return null;
+	}
+
+	return (
+		<div className={ classNames( 'plans-filter-bar-i5__discount-message', { primary: primary } ) }>
+			<div>
+				<span className="plans-filter-bar-i5__discount-message-text">
+					{ isMobile
+						? translate( 'SAVE %(discount)s BY PAYING YEARLY', {
+								args: { discount: highestAnnualDiscount },
+								comment: 'Discount is either a currency-formatted number or percentage',
+						  } )
+						: translate( 'Save %(discount)s', {
+								args: { discount: highestAnnualDiscount },
+								comment: 'Discount is either a currency-formatted number or percentage',
+						  } ) }
+				</span>
+				🎉
```

Comment:
I'm wondering if there is a better way to display this emoji..  Image? Or maybe unicode?  I'll look into this..

Reply:
Seems that using emojis is well supported, from what I read :)

Reply:
Thanks @monsieur-z , yes it appears so, and it looks like there are some other methods that can be used too.  HTML entity `&#xF389;` or JS `String.fromCodePoint(0x1F389)` , But I don't see that one has any advantage over the other in this case. 👍